addons/spectator/functions/fnc_camera.sqf

```diff
@@ -0,0 +1,1081 @@
+        };
+        
+        GVAR(keys) = [];
+        _DIKcodes = true call BIS_fnc_keyCode;
+        _DIKlast = _DIKcodes select (count _DIKcodes - 1);
+        for "_i" from 0 to (_DIKlast - 1) do {
+            GVAR(keys) set [_i, false];
+        };
+        
+        _display = findDisplay 46;
+        
+        GVAR(ehDraw3D) = addMissionEventhandler ["Draw3D", {['Draw3D', _this] call FUNC(draw3D)}];
+        addMissionEventHandler ["Ended", {if (!isNil QGVAR(cam)) then {["Exit"] call FUNC(camera)}}];
+        GVAR(ehKeyDown) = _display displayAddEventHandler ["keyDown", {['KeyDown', _this] call FUNC(camera)}];
+        GVAR(ehKeyUp) = _display displayAddEventHandler ["keyUp", {['KeyUp', _this] call FUNC(camera)}];
+        GVAR(ehMouseButtonDown) = _display displayAddEventHandler ["mouseButtonDown", {['MouseButtonDown', _this] call FUNC(camera)}];
+        GVAR(ehMouseButtonUp) = _display displayAddEventHandler ["mouseButtonUp", {['MouseButtonUp',_this] call FUNC(camera)}];
+        GVAR(ehMouseZChanged) = _display displayAddEventHandler ["mouseZChanged", {['MouseZChanged',_this] call FUNC(camera)}];
+        GVAR(ehMouseMoving) = _display displayAddEventHandler ["mouseMoving", {['Mouse',_this] call FUNC(camera)}];
+        GVAR(ehMouseHolding)  =_display displayAddEventHandler ["mouseHolding", {['Mouse',_this] call FUNC(camera)}];
+
+        //remove mission layer
+        _displayMission = call (uiNamespace getVariable "BIS_fnc_displayMission");
+        _control = _displayMission displayCtrl 11400;
+        _control ctrlSetFade 1;
+        _control ctrlCommit 0;
+
+        //kill layers
+        cutText ["", "Plain"];
+        _layers = missionNamespace getVariable ["BIS_fnc_rscLayer_list", []];
+        for "_i" from 1 to (count _layers - 1) step 2 do {
+            (_layers select _i) cutText ["", "Plain"];
+        };
+        
+        clearRadio;
+        
+        //crosshair
+        _layer = [QGVAR(crosshair)] call BIS_fnc_rscLayer;
+        _layer cutRsc [QGVAR(crosshair), "PLAIN", 2, true];
+        
+        //compass
+        _layer = [QGVAR(compass)] call BIS_fnc_rscLayer;
+        _layer cutRsc [QGVAR(compass), "PLAIN", 2, true];
+        
+        //status
+        _layer = [QGVAR(status)] call BIS_fnc_rscLayer;
+        _layer cutRsc [QGVAR(status), "PLAIN", 2, true];
+        
+        //help
+        _layer = [QGVAR(help)] call BIS_fnc_rscLayer;
+        preloadTitleRsc [QGVAR(help), "PLAIN", 0, true];
+        
+        if (isClass (configFile >> "CfgPatches" >> "ace_nametags")) then {
+            GVAR(tags) = [EGVAR(nametags,showPlayerNames), EGVAR(nametags,showNamesForAI)];
+            EGVAR(nametags,showPlayerNames) = 0;
+            EGVAR(nametags,showNamesForAI) = false;
+        };
+        
+        if (isClass (configFile >> "CfgPatches" >> "ace_interact_menu")) then {
+            [QGVAR(interactCondition), {false}] call EFUNC(common,addCanInteractWithCondition);
```

Comment:
I think a much better way to handle this is with exceptions like SilentSpike said above.